base/ca/src/org/dogtagpki/ct/CTEngine.java

```diff
@@ -0,0 +1,745 @@
+        X509Certificate[] caUnsortedCerts = caCertChain.getChain();
+
+        try {
+            // first, leaf cert;
+            cert.encode(certOut);
+            byte[] certBytes = certOut.toByteArray();
+            certOut.reset();
+            certChain.add(Utils.base64encode(certBytes, false));
+
+            // then add the ca chain, in order (from subCAs to root);
+            X509Certificate[] caSortedCerts = Cert.sortCertificateChain(caUnsortedCerts, true);
+            for (int n = 0; n < caSortedCerts.length; n++) {
+                X509CertImpl caCertInChain = (X509CertImpl) caSortedCerts[n];
+                caCertInChain.encode(certOut);
+                certBytes = certOut.toByteArray();
+                certOut.reset();
+                logger.debug(method + "caCertInChain " + n + " = " +
+                        Utils.base64encode(certBytes, false));
+                certChain.add(Utils.base64encode(certBytes, false));
+            }
+            certOut.close();
+
+            ctRequest.setCerts(certChain);
+            logger.debug(method + " ct_json_request:" + ctRequest.toString());
+        } catch (Exception e) {
+            logger.debug(method + e.toString());
+            throw new EBaseException(e.toString());
+        }
+        return ctRequest;
+    }
+
+    /**
+     * (Certificate Transparency)
+     * certTransSendReq connects to CT host and send ct request
+     * @param ct_host host name
+     * @param ct_port port #
+     * @param ct_uri uri of the CT log server
+     * @param ctReq CT request
+     * @return response content from CT log server
+     */
+    private String certTransSendReq(String ct_host, int ct_port, String ct_uri, CTRequest ctReq) {
+        String method = "CAService.certTransSendReq: ";
+        HttpClient client = new HttpClient();
+        HttpRequest req = new HttpRequest();
+        HttpResponse resp = null;
+
+        logger.debug(method + "begins");
+        try {
+            client.connect(ct_host, ct_port);
+            req.setMethod("POST");
+            req.setURI(ct_uri);
+            req.setHeader("Content-Type", "application/json");
+            req.setContent(ctReq.toString());
+            req.setHeader("Content-Length", Integer.toString(ctReq.toString().length()));
+
+            resp = client.send(req);
+            if (resp == null)
+                return null;
+            logger.debug("version " + resp.getHttpVers());
+            logger.debug("status code " + resp.getStatusCode());
+            logger.debug("reason " + resp.getReasonPhrase());
+            logger.debug("content " + resp.getContent());
+            logger.debug("CAService.certTransSendReq ends");
+        } catch (Exception e) {
+            logger.debug(method + e.toString());
+            return null;
+        }
+
+        return (resp.getContent());
+    }
+
+    /*
+     * (Certificate Transparency)
+     */
+    public static String bytesToHex(byte[] bytes) {
```

Comment:
Minor:  If this is a reusable method to the util package. If specific to SCT, move along with verifySCT()